app/main.py

```diff
@@ -1,6 +1,13 @@
 from fastapi import FastAPI
-from .api import api_router
+from app.api import api_router
+from contextlib import asynccontextmanager
+from app.database import init_db
 
-app = FastAPI()
+@asynccontextmanager
```

Comment:
I want you to focus on the example in the link. Your approach is good but we have a built-in functions to handle the database startup and shutdown.

I will not say to remove this part but please focus on it

One more update, please add try except and close the database if finally statement

Reply:
As I read [here](https://fastapi.tiangolo.com/advanced/events/#lifespan), both approaches—using @app.on_event("startup") and the lifespan function—are valid. However, the main difference is that @app.on_event("startup") is synchronous by default, though it can also be made asynchronous if needed. In contrast, the lifespan function is always asynchronous.

Reply:
Yes, my point is that it will be better to use the built-in function provided by our framework team